src/Metadata/Subscriber/AttributeSubscriberMetadataFactory.php

```
@@ -109,8 +110,23 @@ public function metadata(string $subscriber): SubscriberMetadata
             throw DuplicateSubscribeMethod::mixedWithAll($subscriber);
         }
 
+        $subscriptionId = $subscriberInfo->id;
+
+        if ($subscriptionId === null) {
```

We should rather throw an error message here if the ID has been defined twice, regardless of whether it's on two properties or once via a subscriber attribute and property.